File: Data Structures & Algorithms/products-of-array-discluding-self/submission-2.py
class Solution:
    def productExceptSelf(self, nums: List[int]) -> List[int]:
        res = [0] * len(nums)

        for i in range(len(nums)):
            cur_prod = 1
            for j in range(len(nums)):
                if i == j:
                    continue
                cur_prod *= nums[j]

            res[i] = cur_prod

        return res

File: Data Structures & Algorithms/longest-consecutive-sequence/submission-2.cpp
class Solution {
public:
    int longestConsecutive(vector<int>& nums) {
        int res{};
        std::unordered_set<int> unique_nums(nums.begin(), nums.end());

        for (const int num : unique_nums)
        {
            if (unique_nums.contains(num - 1)) { continue; }

            // Making it here means this num is currently the smallest of the sequence
            int seq_len = 1;
            int cur_num = num;
            while (unique_nums.contains(cur_num + 1))
            {
                seq_len += 1;
                cur_num += 1;
            }

            res = std::max(seq_len, res);
        }

        return res;
    }
};
